File: frontend/src/lib/tauri-api/workspace-attention.ts
import type { WorkspaceAttention } from '../../types/workspace-attention';
import { invokeCommand } from './client';

export function listWorkspaceAttention(): Promise<WorkspaceAttention[]> {
  return invokeCommand<WorkspaceAttention[]>('list_workspace_attention');
}

export function markWorkspaceAttentionRead(workspaceId: string): Promise<void> {
  return invokeCommand<void>('mark_workspace_attention_read', { workspaceId });
}

File: src-tauri/src/commands/workspace_attention.rs
use tauri::State;

use crate::models::WorkspaceAttention;
use crate::services::workspace_attention_service;
use crate::state::AppState;

#[tauri::command]
pub fn list_workspace_attention(
    state: State<'_, AppState>,
) -> Result<Vec<WorkspaceAttention>, String> {
    workspace_attention_service::list_workspace_attention(&state)
}

#[tauri::command]
pub fn mark_workspace_attention_read(
    state: State<'_, AppState>,
    workspace_id: String,
) -> Result<(), String> {
    workspace_attention_service::mark_workspace_attention_read(&state, &workspace_id)
}
